mlrun/datastore/utils.py

```diff
@@ -180,3 +181,53 @@ def get_kafka_brokers_from_dict(options: dict, pop=False) -> typing.Optional[str
             FutureWarning,
         )
     return kafka_bootstrap_servers
+
+
+def transform_list_filters_to_tuple(additional_filters):
+    tuple_filters = []
+    if not additional_filters:
+        return tuple_filters
+    validate_additional_filters(additional_filters=additional_filters)
+    for additional_filter in additional_filters:
+        tuple_filters.append(tuple(additional_filter))
+    return tuple_filters
+
+
+def validate_additional_filters(additional_filters):
+    nan_error_message = "using NaN in additional_filters is not supported"
+
+    if not additional_filters:
+        return
+    for filter_tuple in additional_filters:
+        if not filter_tuple:
+            continue
+        if not isinstance(filter_tuple, (list, tuple)):
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f"mlrun supports additional_filters only as a list of tuples."
+                f" Current additional_filters: {additional_filters}"
+            )
+        if len(filter_tuple) != 3:
+            if all(isinstance(element, (list, tuple)) for element in filter_tuple):
+                raise mlrun.errors.MLRunInvalidArgumentError(
+                    f"mlrun supports additional_filters only as a list of tuples."
+                    f" Current additional_filters: {additional_filters}"
+                )
+            else:
```

Comment:
This doesn't seem to make sense. If all elements are lists or tuples, then we raise an error that "mlrun supports additional_filters only as a list of tuples"? It seems like this whole case can be dropped.
```suggestion
```

Reply:
additional filters does not support nested list inside filter tuples.  Current filter_tuple: {filter_tuple}.